### PR TITLE
Register authorizers check

### LINKED ISSUE #1602: Register authorizers
We should change `addAuthorier`
- This transaction type can only be executed by sc owner, should contain similar check
```
	if err := smartcontractinterface.AuthorizeWithOwner("register", func() bool {
		return gn.ZCNSC.OwnerId == t.ClientID
	}); err != nil {
		return "", err
	}
```
- this command should be added to zwalletcli, something like `auth-register` with operational wallet_id provided to it

### PR DESCRIPTION
**Pull request by uneet7**
## Fixes
#1602 

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/zcnsc/authorizer.go [hunk 1]
```diff
@@ -65,8 +65,11 @@ func (zcn *ZCNSmartContract) AddAuthorizer(
 		return "", common.NewError(code, "authorizer's delegate_wallet not set")
 	}
 
-	if authorizerID != params.StakePoolSettings.DelegateWallet {
-		return "", common.NewError(code, "access denied, allowed for delegate_wallet owner only")
+	// transaction type can only be executed by sc owner
+	if err := smartcontractinterface.AuthorizeWithDelegate("register-authorizer", func() bool {
+		return authorizerID != params.StakePoolSettings.DelegateWallet
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/zcnsc/authorizer.go]: delete this check as we already have one

**Review by dabasov — CHANGES_REQUESTED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

File: code/go/0chain.net/smartcontract/zcnsc/authorizer.go [hunk 2]
```diff
@@ -77,10 +73,14 @@ func (zcn *ZCNSmartContract) AddAuthorizer(
 		return "", err
 	}
 
-	// Validating StakePoolSettings against GlobalNode settings
+	// transaction type can only be executed by sc owner
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/zcnsc/authorizer.go]: The comment should be 'only sc owner can add new authorizer'. In another way to say is that this transaction will be executed only when it is created and signed by the sc owner. Transactions are executed by miners, not the sc owner.

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Review by peterlimg — APPROVED** [review]
lgtm, please fix the conflicts

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | 1602-register-authorizers-check  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/smartcontract/zcnsc/authorizer.go [hunk 3]
```diff
@@ -56,12 +52,23 @@ func (zcn *ZCNSmartContract) AddAuthorizer(
...
+	authorizerID = params.ID
+
 	if params.PublicKey == "" {
 		err = common.NewError(code, "public key was not included with transaction")
 		Logger.Error("public key error", zap.Error(err))
 		return "", err
 	}
 
+	publicKeyBytes, _ := hex.DecodeString(params.PublicKey)
+	clientId = encryption.Hash(publicKeyBytes)
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/zcnsc/authorizer.go:72]: We don't need to get clientID from publicKey as the txn.ClientID has already be verified when the txn is collected.

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**